### DISCUSSION TITLE
on Angular 11 + Ionic 5.30.x version not showing model window

### DISCUSSION THREAD

**Discussion by hasmikak** [NiklasMerz/cordova-plugin-fingerprint-aio#314] [Q&A]
Hi I am trying the latest version with Latest angular and ionic versions. it says Fingerprint option available but not showing the model window.

could you please let me know if any one tried and have solution?

thank you
-Jana

**Comment by NiklasMerz** [comment]
We need more information like Android or iOS, test devices, errors logs etc.

Please try to check logcat on Android or the Xcode console on iOS:

**Comment by hasmikak** [comment]
[](url)
[aio-log.txt](https://github.com/NiklasMerz/cordova-plugin-fingerprint-aio/files/5798583/aio-log.txt)


Here are the logs i have from the device. Samsung S8+. and i have added following  code in the Init of the appComponent.ts

I am getting two alerts.. first Then Alert and immedialty .. gettimg catch Alert.

please let me know if i am missing antyhing.


    
    this.faio.isAvailable().
    then(() => {
      alert('On Init - Finger Print Avalable');
      let options: FingerprintOptions;
      options.cancelButtonTitle = 'Cancel';
      options.description = 'Please authenticate';
      options.fallbackButtonTitle = 'User Backup';
      options.subtitle = 'Coolest Plugin ever';
      options.title = 'Biometric Authentication';
      options.disableBackup = true;

      this.faio.show(
        options
      )
    .then((result: any) => {
      console.log(result)
      alert('On Init -  Biometric authenticated');
    })
    .catch((error: any) => console.log(error));
  
    })
    .catch(() => {
      alert('On Init -  App Component: Finger Print Not Avalable');
    })

**Comment by hasmikak** [comment]
Thank you for quick response.

**Comment by hasmikak** [comment]
here is more info.. attaching package json and code snippet where i am showing the model window. 
and logcat logs. 

I am trying to run on Samsung 8+ android. 


  this.faio.isAvailable().
    then(() => {
      // alert('On Init - Finger Print Avalable');
      let options: FingerprintOptions = {
      cancelButtonTitle: 'Cancel',
      description: 'Please authenticate',
      fallbackButtonTitle: 'User Backup',
      subtitle: 'Coolest Plugin ever',
      title: 'Biometric Authentication',
      disableBackup: true
        }

      this.faio.show(
        options
      )
    .then((result: any) => {
      console.log('On Init -  Biometric authenticated' + result);
      alert('On Init -  Biometric authenticated');
    })
    .catch((error: any) => {
      console.log('On Init - Show Catch  App Component: Finger Print Not Avalable : ' + error);
      alert('On Init - Show Catch  App Component: Finger Print Not Avalable' + error);
    });
  
    })
    .catch((error: any) => {
      console.log('On Init -  App Component: Finger Print Not Avalable' + error);
![WhatsApp Image 2021-01-11 at 9 16 04 PM](https://user-images.githubusercontent.com/72677952/104261460-98f2cd80-5453-11eb-8ffe-4027095f3337.jpeg)

      alert('On Init -  App Component: Finger Print Not Avalable' + error);
    })






[pkg-jsn.txt](https://github.com/NiklasMerz/cordova-plugin-fingerprint-aio/files/5799368/pkg-jsn.txt)

**Comment by hasmikak** [comment]
[catlog.txt](https://github.com/NiklasMerz/cordova-plugin-fingerprint-aio/files/5799371/catlog.txt)

**Comment by NiklasMerz** [comment]
Like the error says you are missing a parameter. Please check the examples again. Also check the Chrome console for errors

You are creating an options object but never assign a value to it. Using `let options: FingerprintOptions;` does not initializes the variable properly.

**Reply by hasmikak** [reply]
this is what i am doing.. 
On Console i dont see any error.. i am passing all the parameters.. 
as per this interface.
but still says missing parameters.

let options: FingerprintOptions = {
cancelButtonTitle: 'Cancel',
description: 'Please authenticate',
fallbackButtonTitle: 'User Backup',
subtitle: 'Coolest Plugin ever',
title: 'Biometric Authentication',
disableBackup: true
}

**Reply by NiklasMerz** [reply]
Are you really sure you are on the latest version? Try removing the android platform and adding it again.

**Reply by hasmikak** [reply]
Thank you for quick reply. Ya I am on.. Latest, let me removing android and try again.

**Reply by hasmikak** [reply]
Is there any way I can directly chat with you?
So that might speed up. Please let me know. I am kind of stuck.

appreciate your help.

**Reply by hasmikak** [reply]
mean while i am trying your suggesion..

**Reply by hasmikak** [reply]
I tried removing platform android and adding back again.. still i see missing parameter message.

**Comment by hasmikak** [comment]
I have another update here.. Today I have created fresh Ionic conference app and installed the fio plugin and tried

still i see an error..  please check this log. i have used the same code from other app to call fio.show().  

is there anything else i need to do in the app after installing the pluin and adding the fio.show code.

please let me know.

[fresh-ionic-app-log.txt](https://github.com/NiklasMerz/cordova-plugin-fingerprint-aio/files/5804914/fresh-ionic-app-log.txt)

**Comment by hasmikak** [comment]
also one request if you have any working sample with latest ionic and latest  aio plugin. please share with me i will test that and see whats the difference.

thank you So much.

**Comment by hasmikak** [comment]
Hi , Good news is.. After removing the plugin completely and added again and i am able to see the Finger print scan option now. 

seems when we directly add plugin its not updating the installed one. could be npm bug.


and Could you please suggest me for the following. 

I would need to store the finger print and authenticate next time when use tries to login. 

what would be the right approach using this package?

appreciate your time helping me so far.

thank you
-Jana

**Comment by NiklasMerz** [comment]
If you want to update a plugin remove it first and install the latest version after.

You don't need to store the fingeprint. The plugin will return success if the fingerprint matches the ones stored in the system settings for unlocking the device.

**Reply by hasmikak** [reply]
oh ok Thank you ..

I would like to store some data based on user logged in using finger print. 

so how to make that differentiate?

any help please.?

thank you

**Reply by NiklasMerz** [reply]
Devices usually only allow one user to store their fingerprints. I don't really understand the question.

**Comment by pearlhunter** [comment]
I can only recommend asking here [https://scand.com/technologies/angular/](https://scand.com/technologies/angular/) as well, I think they could help you.
